### PR TITLE
FileList to new design

### PR DESCRIPTION
**Pull request by reno1979**
- FileList to new design
- FileList updated story
- FileList added testing stubs
- FileUpload to new design
- FileUpload updated react-dropzone dep [breaking change] 
- Storybook added mock request addon

https://www.figma.com/file/4a6C5fMGJ8dd7ZQgdLVcnf/AIP_gemeenteAmsterdam?node-id=683%3A16224

### CODE REVIEW

File: src/form/FileUpload/FileList/FileList.test.tsx [hunk 1]
```diff
@@ -1,44 +1,15 @@
 import React from 'react';
-import { fireEvent, render } from '@testing-library/react';
-import FileList, { Props } from './FileList';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import FileList, { FileListProps } from './FileList';
 import { CustomFile } from '../hooks';
+import { filesWithRejection } from './__stubs__/files';
```
Comment (fortunato) [inline src/form/FileUpload/FileList/FileList.test.tsx:6]: Neat :ok_hand:

File: src/form/FileUpload/FileList/FileList.tsx [hunk 2]
```diff
@@ -74,49 +114,71 @@ const FileListItem: React.FC<FileListItemProps> = ({
...
+	const isIndeterminate = isFileUploadingIndeterminate(file);
+	const { read, readerCleanup } = useFileReader(setPreview);
+
+	useEffect(() => {
+		return () => {
+			readerCleanup();
+		};
+	}, []);
+
+	useEffect(() => {
```
Comment (fortunato) [inline src/form/FileUpload/FileList/FileList.tsx:130]: May I suggest this useEffect is moved into the `useFileReader` hook? I think that would make the implementation a lot cleaner and more reusable.

File: src/form/FileUpload/FileList/FileListStyles.ts [hunk 3]
```diff
@@ -1,71 +1,71 @@
 import styled from 'styled-components';
-import { Button, Icon, themeColor, themeSpacing } from '@amsterdam/asc-ui';
+import { Button, themeColor, themeSpacing } from '@amsterdam/asc-ui';
+
+export const FileListTitleStyle = styled.h6`
+	align-self: baseline;
+	font-weight: 800;
+	font-size: 18px;
+	line-height: 130%;
```
Comment (fortunato) [inline src/form/FileUpload/FileList/FileListStyles.ts:8]: To my knowledge we're not using % line-height values anywhere. Why not pixels like we do elsewhere?

File: src/form/FileUpload/FileList/FileListStyles.ts [hunk 4]
```diff
@@ -1,71 +1,71 @@
...
-export const FileRemoveStyle = styled(Button)`
-	margin-left: auto;
-`;
-
-export const FileCancelStyle = styled(Button)`
+export const FileListItemButtonStyle = styled(Button)`
+	align-self: center;
+	font-weight: 400;
+	font-size: 18px;
+	line-height: 160%;
```
Comment (fortunato) [inline src/form/FileUpload/FileList/FileListStyles.ts:60]: Why %?

File: src/form/FileUpload/FileUploadStyles.ts [hunk 5]
```diff
@@ -21,45 +9,57 @@ export const FileUploadStyle = styled.div`
...
+	justify-content: center;
+	align-items: center;
+
+	& > span {
+		margin-left: ${themeSpacing(2)};
+		padding-right: 0;
+		max-width: ${themeSpacing(4)};
+	}
+
+	${FileListStyle}:not(:empty) ~ ${FileUploadPlaceholderStyle},
```
Comment (fortunato) [inline src/form/FileUpload/FileUploadStyles.ts:61]: This deserves a comment IMHO; I can't read this without diving in.

File: src/form/FileUpload/hooks.ts [hunk 6]
```diff
@@ -64,19 +69,21 @@ export const useFileUpload = (
...
+					if (status >= 200 && status < 400) {
 						// The request has been completed successfully
 						customFile.progress = 100;
-						customFile.response = xhr.responseText;
+						customFile.response = response;
 						customFile.uploadXhrError = undefined;
 						// Let application track uploaded files
 						onFileSuccess && onFileSuccess(customFile);
 					} else {
 						// Something went wrong with the request
```
Comment (fortunato) [inline src/form/FileUpload/hooks.ts:84]: Does this mean that onreadystatechange is never called with the default `0` status without it being an error?

File: src/utils/__mocks__/isBase64UrlImage.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+export const isBase64UrlImage = async (base64String: string) => {
```
Comment (fortunato) [inline src/utils/__mocks__/isBase64UrlImage.ts:1]: Matter of taste, but perfect candidate for a one-liner IMHO

File: src/utils/isBase64UrlImage.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+export const isBase64UrlImage = async (base64String: string) => {
```
Comment (fortunato) [inline src/utils/isBase64UrlImage.ts:1]: I'm missing a return type here. The await on the returned Promise makes it a bit of a head scratcher I think.

**Review by fortunato — CHANGES_REQUESTED** [review]
Nice work. A few minor comments/questions and one request for change regarding the use of `useEffect` in relation to the `useFileReader` hook.

File: src/common/src/Modal/ModalContent/ModalContent.test.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import ModalContent from './ModalContent';
+
+describe('<ModalContent />', () => {
+	it('should remove the margin from the first and last element', () => {
```
Comment (HeeChanVanDerHaar) [inline src/common/src/Modal/ModalContent/ModalContent.test.tsx:6]: Not sure what you want to test here. Don't see any dynamic input/values which might change the margins. I would opt for a more explicit approach. Probably the margins are changing based on a boolean, but I don't see that logic anywhere

File: src/form/FileUpload/FileList/FileList.tsx (hunk 2 above)
Comment (reno1979) [reply src/form/FileUpload/FileList/FileList.tsx:130]: good tip thx

File: src/form/FileUpload/FileList/FileList.tsx [hunk 10]
```diff
@@ -1,70 +1,110 @@
...
+					file={file}
+					fileUploadErrorLabel={fileUploadErrorLabel}
+					fileUploadInProgressLabel={fileUploadInProgressLabel}
+				/>
+			))}
+		</FileListStyle>
+	);
+};
+
+const useFileReader = (setPreview: (val: string) => void) => {
```
Comment (HeeChanVanDerHaar) [inline src/form/FileUpload/FileList/FileList.tsx:81]: Would it be a bit cleaner just have `file` as a dependency/param? And the hook just exposes the result? So you have all the  filereader logic in the hook instead in the implementation side? Something like
`const {result: preview} = useFileReader(file)`.
Would also move it to it's own file, so you can test it :)

File: src/form/FileUpload/hooks.ts (hunk 6 above)
Comment (reno1979) [reply src/form/FileUpload/hooks.ts:84]: No the state change is called on every state change, we only handle the state change when it has the state DONE.

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/status

**Pull request merged**